src/controllers/RequestController.js

```diff
@@ -66,6 +66,45 @@ class RequestController {
       return HelperMethods.serverError(res);
     }
   }
+
+  static async editARequest(req, res) {
+    try {
+      const { id } = req.decoded;
+      const {
+        requestId, origin, destination, flightDate, returnDate, reason,
+      } = req.body;
+
+      const requestExist = await Request.findOne({
+        where: {
+          id: requestId,
+          userId: id,
+        }
+      });
+      if (requestExist) {
+        if (requestExist.dataValues.status === 'pending') {
+          if (requestExist.dataValues.returnDate) {
+            const convertFlightDate = new Date(flightDate).toISOString();
+            const convertReturnDate = new Date(returnDate).toISOString();
+            if (convertFlightDate > convertReturnDate) return HelperMethods.clientError(res, 'Invalid Date Parameters', 400);
+          }
+
+          const updatedRequest = await requestExist.update({
+            origin, destination, flightDate, returnDate, reason
```

Comment:
instead of passing each field in the req.body, just pass in `req.body`. because if we have more columns in the request table, then we have to come here to edit it. Also what if its only one column we just want to update? then this means we will have `undefined` for other columns.